searx/plugins/__init__.py

```diff
@@ -77,6 +87,80 @@ def call(self, ordered_plugin_list, plugin_type, request, *args, **kwargs):
         return ret
 
 
+def load_external_plugins(plugin_names):
+    plugins = []
+    for name in plugin_names:
+        logger.debug('loading plugin: {0}'.format(name))
+        try:
+            pkg = import_module(name)
+        except Exception as e:
+            logger.critical('failed to load plugin module {0}: {1}'.format(name, e))
+            exit(3)
+
+        pkg.__base_path = dirname(abspath(pkg.__file__))
+
+        prepare_package_resources(pkg, name)
+
+        plugins.append(pkg)
+        logger.debug('plugin "{0}" loaded'.format(name))
+    return plugins
+
+
+def sync_resource(base_path, resource_path, name, target_dir, plugin_dir):
+    dep_path = join(base_path, resource_path)
+    file_name = basename(dep_path)
+    resource_path = join(target_dir, file_name)
+    if not exists(resource_path) or sha_sum(dep_path) != sha_sum(resource_path):
+        try:
+            copyfile(dep_path, resource_path)
+        except:
```

Comment:
If we do not catch exceptions, they go upwards and the effect is the same, the application exits .. by example, if you drop the try/catch block you get:

````
FileNotFoundError: [Errno 2] No such file or directory: '/800GBPCIex4/share/searx/local/py3/lib/python3.8/site-packages/plugin1/resources/bg.js'
````